Task_4.1/Server.cs

```diff
@@ -0,0 +1,91 @@
+using System.Runtime.CompilerServices;
+
+namespace Task_4._1;
+
+using System.IO;
+
+public class Server
+{
+    public Server(int address, int portsCount, string rootDirectory) //(int address, string rootDirectory)
+    {
+        Address = address;
+        _rootDirectory = rootDirectory;
+        _ports = new bool[portsCount];
+    }
+
+    public readonly int Address;
+    private readonly bool[] _ports; // true -> порт занят, false -> порт свободен
+    private readonly string _rootDirectory;
+
+    public bool TakePort(int port)
+    {
+        if (port < 0 || port >= _ports.Length || _ports[port])
+        {
+            return false;
+        }
+
+        _ports[port] = true;
+        return true;
+    }
+
+    public void ReleasePort(int port)
+    {
+        if (port >= 0 && port < _ports.Length)
+        {
+            _ports[port] = false;
+        }
+    }
+
+    private bool _isDirectory(string path) =>
+        Directory.Exists(Directory.GetDirectoryRoot(_rootDirectory)[1..] + _rootDirectory + path) ||
+        Directory.Exists(Directory.GetDirectoryRoot(path)[1..] + path);
+
+    public Tuple<int, List<ListAnswer>> List(string path)
```

Comment:
```suggestion
    public (int size, List<ListAnswer> answers) List(string path)
```
Так и читаемее код, и понятнее, что за возвращаемые значения